runtime/lua/vim/treesitter.lua

```diff
@@ -187,15 +187,20 @@ end
 ---@returns string
 local function buf_range_get_text(buf, range)
   local start_row, start_col, end_row, end_col = M._range.unpack4(range)
+  local insert_nl = false
   if end_col == 0 then
     if start_row == end_row then
       start_col = -1
       start_row = start_row - 1
     end
     end_col = -1
     end_row = end_row - 1
+    insert_nl = true
   end
```

Comment:
It looks like we could also get rid of this `if end_col == 0 then ... end` block completely to achieve the same behavior, and use directly return what `nvim_buf_get_get_text()` returns.

That said, one might wonder why `get_node_text` was designed in the current way. It was introduced 58bbc2ea0b3dfed13471e8cc0447d7598be24276 --- not sure why. Maybe to match the behavior as the initial implementation 1f3c0593eb1d4e54ce1edf35da67d184807a9280 where only rows are specified.

Reply:
I ack the message. I will have a deeper look this week. Thanks for the review.

Reply:
I think this check prevents `Index out of bounds` errors when the node as at the end of the file. I just encountered this myself trying to get text using only `nvim_buf_get_text` without any special magic. So it looks like the check is necessary